CMakeLists.txt

```diff
@@ -76,11 +76,13 @@ check_symbol_exists(strlcpy "string.h" HAVE_STRLCPY)
 
 if(HAVE_STRLCPY)
 	message(STATUS "Existing strlcpy found, will *not* use local definition by setting -DHAVE_STRLCPY.")
-	add_definitions(-DHAVE_STRLCPY)
 else()
 	message(STATUS "No strlcpy found, will use local definition")
 endif()
 
+configure_file(userspace/common/common_config.h.in ${PROJECT_BINARY_DIR}/common/common_config.h)
```

Comment:
I like the approach of moving the generated header file into the `${PROJECT_BINARY_DIR}/` directory, is should make it much easier to cleanup, even if it does lead to some extra configuration needed in cmake.